analyze_data.py

```diff
@@ -115,7 +130,11 @@ def get_all_data_as_dict(filename):
         dict: Dictionary with all data from parse_filename and
               get_data_from_file
     """
-    return {}
+    filedata = parse_filename(filename)
+    data = get_data_from_file(filename)
+    data_dictionary = filedata.copy()
+    data_dictionary.update(data)
+    return {data_dictionary}
 
 
 def get_all_files_in_folder(folder):
```

Comment:
Try to get this from the "folders" arg

What is the return of os#listdir()?  What is its type?

```
In [2]: os.listdir?
Docstring:
listdir(path) -> list_of_strings

Return a list containing the names of the entries in the directory.

    path: path of directory to list

The list is in arbitrary order.  It does not include the special
entries '.' and '..' even if they are present in the directory.
Type:      builtin_function_or_method
```

It is already a list so now you are returning a list with one element which is a list, instead of just a list of filenames.